### PR TITLE
[exporter/datadog] add example k8s configs

### PR DESCRIPTION
**Pull request by ericmustin**
**Description:** <Describe what has changed.>

Adds an example k8s manifest for deploying the collector as agent and as collector in a k8s environment, with datadog specific processors and exporters enabled.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by ericmustin** [comment]
would like to have this reviewed by colleagues at datadog before merging, thanks! cc @mx-psi

File: exporter/datadogexporter/example/example_k8s_manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,257 @@
+# This manifest file is meant as an example of how to deloy otel-agent's as daemonset and otel-collector as a standalone service
```
Comment (mx-psi) [inline exporter/datadogexporter/example/example_k8s_manifest.yaml:1]:
```suggestion
# This manifest file is meant as an example of how to deploy otel-agent's as daemonset and otel-collector as a standalone service
```

File: exporter/datadogexporter/example/example_k8s_manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+# In a way the correctly identifies hostnames for individual k8s nodes. This is meant as an example only and may differ depending
```
Comment (mx-psi) [inline exporter/datadogexporter/example/example_k8s_manifest.yaml:2]: Something is missing in this sentence, right?

**Review by mx-psi — CHANGES_REQUESTED** [review]

**Comment by jpkrohling** [comment]
Do you mind if I ask why this component must have a Kubernetes example? We have Kubernetes examples in the main repository, and no individual component has one, to my knowledge. Nothing really against it, just trying to understand if there's something special about this component.

**Comment by ericmustin** [comment]
@jpkrohling no worries at all, don't mind you asking. So, we've been working on onboarding documentation for datadog's docs (this is public, feel free to take a look: https://github.com/DataDog/documentation/pull/9529) , and noticed that a lot of users of otel at datadog are new to both otel and datadog at the same time, and so documenting things in a way that is just an incremental "add these exporters or processors to an existing setup" is confusing because there is no existing setup.

Additionally, the current k8s docs we've found to be not a wonderful fit for capturing metadata that is  especially important for a vendor like ddog that is doing host count-based billing. Specifically there's no node or pod level information captured here via a `k8s_tagger` processor: https://github.com/open-telemetry/opentelemetry-collector/blob/master/examples/k8s/otel-config.yaml . We additionally have some preferences around the `batch` processor that may not be something everyone would want setup globally.

So, all those things combined, it's just far  easier and simpler to have a single doc that helps users onboard to datadog and OpenTelemetry simultaneously. And we'd prefer that doc not be `in-line` markdown within our docs site, but hosted in the otel-collector-contrib repo so folks can find it more easily if they're coming from within the `contrib` repo instead of via our docs site.

Additionally, when we have our vendor specific docs in a good place and verified with users, we'd like to contribute back upstream to the main docs repo for otel. we've found that the docker examples and k8s examples there have room for improvement.  Once those are in a better place we'll try to migrate to using those, but in the meantime  we'd prefer to have strong examples within the `datadogexporter` repo.

For context, we've found there are a number of other vendors that are taking a similar approach to what we're doing, but choosing to keep their examples and docs proprietary or hosted largely within their own repos. We feel the approach we're taking is more community friendly.

Let me know if that sounds reasonable or if there's anything we can  provide here to improve the comfort level around this.

**Comment by jpkrohling** [comment]
Thanks for the explanation, sounds good to me.

> we've found that the docker examples and k8s examples there have room for improvement

+1, I have a blog post draft ready to be submitted on this topic, as I don't think the examples are satisfactory.

File: exporter/datadogexporter/example/example_k8s_manifest.yaml (hunk 2 above)
Comment (ericmustin) [reply exporter/datadogexporter/example/example_k8s_manifest.yaml]: ah yup, updated.

**Review by mx-psi — APPROVED** [review]
LGTM, we should open the issues needed for these docs to be updated in the k8s_tagger and in the OpenTelemetry docs.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**